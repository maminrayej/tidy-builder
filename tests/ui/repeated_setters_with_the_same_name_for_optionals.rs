#[derive(tidy_builder::Builder)]
pub struct MyStruct {
    #[builder(each = "optional_args")]
    optional_args: Option<Vec<String>>,
}

fn main() {
    let my_struct = MyStruct::builder().optional_args(vec![]).build();
}
