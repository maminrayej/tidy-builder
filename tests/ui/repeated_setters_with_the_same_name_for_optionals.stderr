error[E0308]: mismatched types
 --> tests/ui/repeated_setters_with_the_same_name_for_optionals.rs:8:55
  |
8 |     let my_struct = MyStruct::builder().optional_args(vec![]).build();
  |                                                       ^^^^^^ expected struct `String`, found struct `Vec`
  |
  = note: expected struct `String`
             found struct `Vec<_>`
  = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
